setup/action.yml

```
@@ -0,0 +1,26 @@
+name: setup-gh
+description: Setup Notation CLI on GitHub Action runners
+branding:
+  icon: check-circle
+  color: blue
+inputs:
+  version:
+    description: Version of Notation CLI to install
+    required: false
+    default: 1.0.0-rc.7
+  key_name:
+    description: Name of the signing key for Notation
+    required: false
+    default: self-signed
+  certificate_key_id:
+    description: Key identifier for the signing certificate
+    required: false
+  plugin_name:
+    description: Name of the Notation plugin to install
+    required: false
+  plugin_version:
+    description: Version of the Notation plugin to install
+    required: false
```

These inputs are related to `sign` and/or `verify`, I think we should remove them from the `setup` action. 
In addition, we should add a new input called `url` for installing notation CLI via a URL. For example, if I want to build a release of Notation CLI on my local branch and use it in the rest of the workflow, the current action cannot help me to do that.
```suggestion
inputs:
  version:
    description: Version of Notation CLI to install
    required: false
    default: 1.0.0-rc.7
  url:
    description: URL of Notation CLI to install
    required: false
```

---

+1

---

> To clarity, all plugin setup is now going to be done by the sign action?

@duffney Yes, `sign` and `verify` would have their own steps of configuring plugins. `setup` will only set up the Notation CLI.

---

I'm uncertain how much value the `url` input will add and how much it will be used. Even from the perspective of a developer working on Notation, I think the chances of them needing to use a dev branch within a workflow are low and the odds of a consumer of the action needing it to be even lower. With that in mind does it add enough value to add in? I could have the wrong perspective and be totally wrong, so please feel free to correct my understanding.